packages/replay-next/src/contexts/FocusContext.tsx

```diff
@@ -80,29 +88,28 @@ export function FocusContextRoot({ children }: PropsWithChildren<{}>) {
   // to indicate that what's currently being showed is stale.
   const [isTransitionPending, startTransition] = useTransition();
 
-  const debouncedSetDeferredRange = useDebouncedCallback(
-    (newRange: TimeStampedPointRange | null) => {
-      startTransition(() => {
-        setDeferredRange(newRange);
-      });
-    },
-    FOCUS_DEBOUNCE_DURATION
-  );
-
   // Refine the loaded ranges based on the focus window.
   useEffect(() => {
-    if (range === null || range === initialRange) {
+    if (rangeForDisplay === null || rangeForDisplay === initialRange) {
       return;
     }
 
-    const timeoutId = setTimeout(() => {
-      client.requestFocusWindow({ begin: range.begin, bias, end: range.end });
+    const timeoutId = setTimeout(async () => {
+      const range = await client.requestFocusWindow({
+        begin: rangeForDisplay.begin,
+        bias,
+        end: rangeForDisplay.end,
+      });
+      setRange(range);
+      startTransition(() => {
+        setRangeForSuspense(range);
+      });
```

Comment:
I think there's a small potential race here, if this effect gets destroyed in between the `await` on line 98 and the state updates on lines 103-106.

I think we need something like this to make it safe:
```js
  let cancelled = false;

  const timeoutId = setTimeout(async () => {
    const range = await client.requestFocusWindow({
      begin: rangeForDisplay.begin,
      bias,
      end: rangeForDisplay.end,
    });

    if (!cancelled) {
      setRange(range);
      startTransition(() => {
        setRangeForSuspense(range);
      });
    }
  }, FOCUS_DEBOUNCE_DURATION);

  return () => {
    cancelled = true;

    clearTimeout(timeoutId);
  };
```